Classes/common/HTTP/CDTURLSessionTask.m

```diff
@@ -84,12 +92,15 @@ - (void)cancel
     if (t) {
         [t cancel];
     }
+    _finished = YES;
```

Comment:
I'm not sure I understand what you mean. This still cancels the `NSURLSessionTask` stored in the `self.inProgressTask`  property, which gets set in `resume`.